### PR TITLE
Add  to selector-no-type.

### LINKED ISSUE #1706: Add ignoreTypes: [] option to selector-no-type
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This is a feature request for `selector-no-type` to be allowed a whitelist of types to be used.

> Which rule, if any, is this issue related to?

`selector-no-type`

> What CSS is needed to reproduce this issue?

e.g.

```css
fieldset:disabled .a-button {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```yaml
extends: stylelint-config-standard
rules:
  selector-no-type: true
```

> Which version of stylelint are you using?

`7.0.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Automated through webpack.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Not initially, but we are using Scss so our source looks more like `fieldset:disabled &` over having the selector directly behind the type.

> What did you expect to happen?

I expect a configuration option to allow us to specifically allow `fieldset` (or any other given elements to the whitelist) to not trigger the `unexpected type error`.

> What actually happened (e.g. what warnings or errors you are getting)?

Unexpected type error on the line defining the disabled fieldset target.

### PR DESCRIPTION
**Pull request by Garbee**
Fixes #1706 

PTAL

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@Garbee Thanks for PR!

File: src/rules/selector-no-type/README.md [hunk 1]
```diff
@@ -79,3 +79,25 @@ The following patterns are *not* considered warnings:
...
+Given:
+
+```js
+["/^my-/", "fieldset"]
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+fieldset .foo { }
```
Comment (jeddy3) [inline src/rules/selector-no-type/README.md]: Can we lose the `.foo` in this examples please.

File: src/rules/selector-no-type/__tests__/index.js [hunk 2]
```diff
@@ -217,3 +217,28 @@ testRule(rule, {
     description: "ignore less interpolation inside @for for nested descendant",
   }],
 })
+
+testRule(rule, {
+  ruleName,
+  config: [ true, { ignoreTypes: [ "fieldset", "/^my-/" ] } ],
+  skipBasicChecks: true,
+
+  accept: [ {
```
Comment (jeddy3) [inline src/rules/selector-no-type/__tests__/index.js]: Can we focus these on *just* the CSS that is being tested e.g.

```js
testRule(rule, {
  ruleName,
  config: [ true, { ignoreTypes: [ "fieldset", "/^my-/" ] } ],
  skipBasicChecks: true,

  accept: [ {
    code: "fieldset {}",
  }, {
    code: "my-type {}",
  }, {
    code: "my-other-type {}",
  } ],

  reject: [ {
    code: "a {}",
    line: 1,
    column: 1,
  }, {
    code: "not-my-type {}",
    message: messages.rejected,
    line: 1,
    column: 1,
  } ],
})
```

Here's we testing:

- 1st: checking string input
- 2nd & 3rd - checking regex input
- 4th - checking simple type selectors still flag
- 5th - checking regex is working

**Comment by jeddy3** [comment]
@Garbee Thanks for the PR :) 

I've made a request for the docs and for the tests.

The [very last thing to do](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules) is add a changelog entry. Can you add one please.

**Comment by Garbee** [comment]
Addressed the comments. Should be good to go as long as the reducing to the single test is alright.

**Comment by jeddy3** [comment]
> Should be good to go as long as the reducing to the single test is alright.

Whoops, I meant swap your tests for the ones below.

```js
testRule(rule, {
  ruleName,
  config: [ true, { ignoreTypes: [ "fieldset", "/^my-/" ] } ],
  skipBasicChecks: true,

  accept: [ {
    code: "fieldset {}",
  }, {
    code: "my-type {}",
  }, {
    code: "my-other-type {}",
  } ],

  reject: [ {
    code: "a {}",
    line: 1,
    column: 1,
  }, {
    code: "not-my-type {}",
    message: messages.rejected,
    line: 1,
    column: 1,
  } ],
})
```

The issue with your original tests is that there were distractions that were irrelevant to the rule e.g. declarations within the blocks and *class selectors* within the selector. The above has no distractions and is the minimum needed to test that the string input works, the regex input works and that the very basic function of the rule still work.

I'll merge this PR and make the change, so we can get this feature out in today's release for you :)

**Comment by Garbee** [comment]
Oh. Oops, I completely mis-understood that feedback then.

Thanks for handling it so quickly! Also, good work on the codebase here, this was *super* easy to add.

**Pull request merged**